lib/codecs/tests/data/avro/avro_generate_fixtures.patch

```diff
@@ -0,0 +1,501 @@
+diff --git a/lib/codecs/Cargo.toml b/lib/codecs/Cargo.toml
```

Comment:
Thanks for adding such extensive tests!

Nit 🥜: Rename `lib/codecs/tests/data/avro/avro` to `lib/codecs/tests/data/avro/generated`.

We really want to move away from this patch practice. A better alternative is to add a new separate target under a `tests/bin` directory. Then we can execute it with a command like `cargo --bin generate-avro-fixtures` which overwrites an existing `lib/codecs/tests/data/avro/generated` directory. Finally, in the `README.md` we can just include this single command and a quick overview of what this generated folder contains and when it is required to regenerate the fixtures.

Reply:
I deleted the patch and use feature to control it, `cargo test --features generate-avro-fixtures --package codecs --test avro -- generate_avro_fixtures` which overwrites an existing lib/codecs/tests/data/avro/generated test data.

Reply:
This is a very good step forward. 

However, I would still prefer this to be a separate bin target to fully decouple it from the rest of the Vector codebase. Also, semantically this is not a test so it might be confusing to future readers/maintainers.

To be more specific, in `Cargo.toml` add:
```
[[bin]]
name = "generate-avro-fixtures"
path = "lib/codecs/tests/bin/generate-avro-fixtures.rs"
```
And in `lib/codecs/tests/bin/generate-avro-fixtures.rs`:
```
fn main() {
    // run all generators
}
```

And finally invoke with:
```
cargo run --bin generate-avro-fixtures
```

Reply:
Is this [chore(avro codec): move fixtures generator to tests/bin](https://github.com/vectordotdev/vector/pull/18636/commits/dc63aeeb56781b567a0410596f275679fa33b5d0) ok ?

Reply:
It's great, thank you.